v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstovectorsearch/BigtableChangeStreamsToVectorSearch.java

```diff
@@ -0,0 +1,235 @@
+                    ? BigtableIO.ExistingPipelineOptions.RESUME_OR_FAIL
+                    : BigtableIO.ExistingPipelineOptions.FAIL_IF_EXISTS)
+            .withProjectId(bigtableProjectId)
+            .withAppProfileId(options.getBigtableChangeStreamAppProfile())
+            .withInstanceId(options.getBigtableReadInstanceId())
+            .withTableId(options.getBigtableReadTableId())
+            .withMetadataTableInstanceId(options.getBigtableChangeStreamMetadataInstanceId())
+            .withMetadataTableTableId(options.getBigtableMetadataTableTableId())
+            .withStartTime(startTimestamp);
+
+    PCollectionTuple results =
+        pipeline
+            .apply("Read from Cloud Bigtable Change Streams", readChangeStream)
+            // TODO(meagar): Remove this
+            .apply("Log Changes", ParDo.of(new LogChangeStreamFn()))
+            .apply("Create Values", Values.create())
+            .apply(
+                "Converting to Vector Search Datapoints",
+                ParDo.of(
+                        new ChangeStreamMutationToDatapointOperationFn(
+                            options.getEmbeddingColumn(),
+                            options.getEmbeddingByteSize(),
+                            options.getCrowdingTagColumn(),
+                            Utils.parseColumnMapping(options.getAllowRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getDenyRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getIntNumericRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getFloatNumericRestrictsMappings()),
+                            Utils.parseColumnMapping(options.getDoubleNumericRestrictsMappings())))
+                    .withOutputTags(
+                        ChangeStreamMutationToDatapointOperationFn.UpsertDatapointTag,
+                        TupleTagList.of(
+                            ChangeStreamMutationToDatapointOperationFn.RemoveDatapointTag)));
+    results
+        .get(ChangeStreamMutationToDatapointOperationFn.UpsertDatapointTag)
+        .apply(WithKeys.of("foo")) // TOOD(meagar): Better name for place-holders?
+        .apply(
+            "Batch Contents",
+            GroupIntoBatches.<String, IndexDatapoint>ofSize(
+                    bufferSizeOption(options.getUpsertMaxBatchSize()))
+                .withMaxBufferingDuration(
+                    bufferDurationOption(options.getUpsertMaxBufferDuration()))
+            // .withShardedKey() // TODO(meagar): What does this do?
+            )
+        .apply("Values", Values.create())
+        .apply(
+            "Upsert Datapoints to VectorSearch",
+            ParDo.of(new UpsertDatapointsFn(vertexEndpoint, indexName)));
+
+    results
+        .get(ChangeStreamMutationToDatapointOperationFn.RemoveDatapointTag)
+        .apply(WithKeys.of("foo")) // TOOD(meagar): Better name for place-holders?
```

Comment:
It wasn't clear to me why we need to introduce keys for GroupIntoBatches to work, and what value I should use for the key since it doesn't actually matter. Can I do something like `WithKeys.of(0)`?